client/js/test/table-view.test.js

```diff
@@ -12,6 +12,46 @@ describe('table-view', () => {
   });
   // jest provides the server-side DOM **
 
+  // describe('add button', () => {
+  //   it('adds row', () => {
+  //     // set up the initial state
+  //     const numCols = 3;
+  //     const numRows = 8;
+  //     const model = new TableModel(numCols, numRows);
+  //     const view = new TableView(model);
+  //     view.init();
+
+  //     // inspect the initial state
+  //     let addRow = document.getElementById('add-row');
+
+  //     // simulate user action
+  //     addRow.click();
+  //     let trs = document.querySelectorAll('TBODY TR');
+
+  //     // inspect the resulting state
+  //     expect(trs.length).toBe(numRows + 1);
+  //   });
+
+  //   it('adds column', () => {
```

Comment:
Also, add more tests that test your new feature functionality. You should develop some code, then write one or more tests for it, then develop more code, then write and run one or more tests for it. That way you can't possibly reach a state of terrible debugging.